jalib/jsocket.cpp

```diff
@@ -152,25 +154,52 @@ jalib::JSockAddr::JSockAddr(const char *hostname /* == NULL*/,
 
 jalib::JSocket::JSocket()
 {
-  _sockfd = jalib::socket(AF_INET, SOCK_STREAM, 0);
+
+	if( jalib::JSocket::isFileSocketDefined() ) {
+
+		memset( &FILE_SOCKET_ADDR, 0, sizeof(struct sockaddr_un) );
+		FILE_SOCKET_ADDR.sun_family = AF_UNIX;
+		strncpy( FILE_SOCKET_ADDR.sun_path, getenv(ENV_VAR_DMTCP_FILE_SOCKET_PATH), sizeof(FILE_SOCKET_ADDR.sun_path) - 1 );
+		_sockfd = jalib::socket(AF_UNIX, SOCK_STREAM, 0);
+	} else {
+    _sockfd = jalib::socket(AF_INET, SOCK_STREAM, 0);
+	}
+}
+
+bool jalib::JSocket::isFileSocketDefined() {
+  bool isDefined = false;
+  if ( getenv( ENV_VAR_DMTCP_FILE_SOCKET_PATH ) != NULL ) {
```

Comment:
Why aren't we validating the contents of the environment variable?

Reply:
We are validating to see if we are using a INET or Unix socket, since each have different function signatures.